### PR TITLE
docs: deno fmt list of files & how they are matched

### PR DESCRIPTION
**Pull request by jthegedus**
This would describe to users at a glance what file types are supported and how the formatter matches those file types.

I do not know if this list is correct, but I came to this page expecting this information to be shared.

### CODE REVIEW

File: tools/formatter.md [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
-JavaScript, Markdown, JSON, and JSONC code.
+Deno ships with a built in code formatter that will auto-format the following files:
+
+| File Type  | Extension          |
+| ---------- | ------------------ |
+| JavaScript | `.js`              |
+| TypeScript | `.ts`              |
+| Markdown   | `.md`, `.markdown` |
+| JSON       | `.json`            |
+| JSONC      | `.jsonc`           |
```
Comment (satyarohith) [inline tools/formatter.md:11]: Also supports JSX (`.jsx` and `.tsx`).

Comment (jthegedus) [reply tools/formatter.md:11]: Would you class that as

|JSX|`.jsx`|
|TSX|`.tsx`|

Or would that fall under JS/TS


|JavaScript|`.js`,`.jsx`|
|TypeScript|`.js`,`.tsx`|

?

Comment (bartlomieju) [reply tools/formatter.md:11]: I suggest the first option

**Comment by jthegedus** [comment]
I performed this update from the GitHub UI and haven't touched the files triggering the `fmt --check` errors. Is it okay to be merged by an admin or do I need to pull everything locally and merge `main` in?

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you.

@jthegedus yes I did format the code, it's fine

**Pull request merged**